compiler/resolution/functionResolution.cpp

```diff
@@ -8928,10 +8928,9 @@ static void insertUnrefForArrayReturn(FnSymbol* fn) {
           call->insertAtTail(unrefCall);
           FnSymbol* unrefFn = resolveNormalCall(unrefCall);
           resolveFns(unrefFn);
-
-          if (unrefFn->retType == tmp->typeInfo()) {
-            // If the types are equal, we must be dealing with a non-view
-            // array, so we can remove the useless unref call.
+          if (unrefFn->where == NULL) {
```

Comment:
I was hoping to find a cleaner way than yet another super-special flag, but at this point a flag may be best.